topics/single-cell/tutorials/EBI-retrieval/tutorial.md

```diff
@@ -0,0 +1,192 @@
+
+# Metadata manipulation
+
+Before creating an AnnData object, we need to make a small modification in experimental design table. The dataset contains information about 7 samples N701 – N707), however in the experimental design table (cell metadata) they are just numbered from 1 to 7. The plotting tool that we will going to use later will fail if the entries are integers and not categoricals, so we will change "1" to "N01" and so on. You can simply preview the experimental design dataset and move to the column "Sample Characteristic[individual]" (that's where the information about batch is - don't worry, we will rename the column header later!). Make a note of the number of that column - number 12 - we will need it to change the batch number to batch name. 
+
+> <hands-on-title> Change batch numbers into names </hands-on-title>
+>
+> 1. Change the datatype of `EBI SCXA Data Retrieval on E-MTAB-6945 exp_design.tsv` to `tabular`:
+>
+>    {% snippet faqs/galaxy/datasets_change_datatype.md datatype="tabular" %}
+>
+> 2. {% tool [Column Regex Find And Replace](toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3) %} with the following parameters:
+>    - *"Select cells from"*: `EBI SCXA Data Retrieval on E-MTAB-6945 exp_design.tsv`
+>    - *"using column"*: `c12`
+>    - In *"Check"*:
+>        - {% icon param-repeat %} *"Insert Check"*
+>            - *"Find Regex"*: `1`
+>            - *"Replacement"*: `N01`
+>        - {% icon param-repeat %} *"Insert Check"*
+>            - *"Find Regex"*: `2`
+>            - *"Replacement"*: `N02`
+>        - {% icon param-repeat %} *"Insert Check"*
+>            - *"Find Regex"*: `3`
+>            - *"Replacement"*: `N03`
+>        - {% icon param-repeat %} *"Insert Check"*
+>            - *"Find Regex"*: `4`
+>            - *"Replacement"*: `N04`
+>        - {% icon param-repeat %} *"Insert Check"*
+>            - *"Find Regex"*: `5`
+>            - *"Replacement"*: `N05`
+>        - {% icon param-repeat %} *"Insert Check"*
+>            - *"Find Regex"*: `6`
+>            - *"Replacement"*: `N06`
+>        - {% icon param-repeat %} *"Insert Check"*
+>            - *"Find Regex"*: `7`
+>            - *"Replacement"*: `N07`
+>
+> 4. Rename {% icon galaxy-pencil %} output `Cell metadata`
+> 
+{: .hands_on}
+
+Now we can create an AnnData object!
+
+# Creating AnnData object
+
+> <hands-on-title> Task description </hands-on-title>
+>
+> 
+> 1. {% tool [Scanpy Read10x](toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_read_10x/scanpy_read_10x/1.8.1+galaxy9) %}
+> 2. Make sure you are using version **1.8.1+galaxy9** of the tool (change by clicking on {% icon tool-versions %} Versions button):
```

Comment:
@wee-snufkin what is broken in the latest version?

Should we hide this latest version?

Reply:
That's the error message when I'm running this step with 1.8.1+galaxy93
![image](https://github.com/galaxyproject/training-material/assets/44121095/f0f4e77a-87dc-4974-9ef2-d27b25b2e7db)

If it's not a quick fix, we might want to hide the latest version

Reply:
This error is fixable. We just need an older numpy version. e.g. numpy==1.21

Can you try this on EU please with the latest version? If this works I will fix the upstream package.

Reply:
@pcm32 This is a matter of tool developer skills, if I'm not mistaken? Are you able to advise here?

Reply:
We don't seem to get this error on instances using singularity (although not sure at the exact wrapper version). Those newer versions only have changes in the galaxy wrapper, the dependency had remained the same for a long time I think

Reply:
So what's going on with this? I'm reviewing this following @wee-snufkin instructions in the tutorial to use the older version. If this issue gets fixed, we can update it I guess

Reply:
I've copied this discussion over to here so we don't lose it in the merged PR: https://github.com/galaxyproject/tools-iuc/issues/5636